### PR TITLE
Minor refactor of code accessing the eval map. 

### PR DESCRIPTION
**Pull request by pleath**
Let all callers get the benefit of debug checks, etc.

### CODE REVIEW

**Comment by pleath** [comment]
@mrkmarron is this correct for the call from NSSnapValues::InflateTopLevelEvalFunctionBodyInfo? (Thanks.)

**Comment by mrkmarron** [comment]
Yes, this should be correct as the guard is `IsTTDRecordOrReplayModeEnabled` which is true if the runtime is started in record or replay mode (even if recording/replaying is temporarily disabled). Clearly if we are in inflating a snapshot we should already be in replay mode.

**Comment by mrkmarron** [comment]
Otherwise LGTM from a TTD standpoint.

File: lib/Runtime/Base/ScriptContext.cpp [hunk 1]
```diff
@@ -2293,7 +2293,30 @@ namespace Js
         return success;
     }
 
-    void ScriptContext::AddToEvalMap(FastEvalMapString const& key, BOOL isIndirect, ScriptFunction *pFuncScript)
+    void ScriptContext::AddToEvalMap(FastEvalMapString const& key, BOOL isIndirect, ScriptFunction *pfuncScript)
```
Comment (agarwal-sandeep) [inline lib/Runtime/Base/ScriptContext.cpp]: LGTM

**Pull request merged**